packages/flutter/lib/src/widgets/tap_region.dart

```diff
@@ -0,0 +1,534 @@
+      markNeedsLayout();
+    }
+  }
+
+  /// The registry that this [RenderTapRegion] should register with.
+  ///
+  /// If the `registry` is null, then this region will not be registered
+  /// anywhere, and will not do any tap detection.
+  ///
+  /// A [RenderTapRegionSurface] is a [TapRegionRegistry].
+  TapRegionRegistry? get registry => _registry;
+  TapRegionRegistry? _registry;
+  set registry(TapRegionRegistry? value) {
+    if (_registry != value) {
+      if (_isRegistered) {
+        _registry!.unregisterTapRegion(this);
+        _isRegistered = false;
+      }
+      _registry = value;
+      markNeedsLayout();
+    }
+  }
+
+  @override
+  void layout(Constraints constraints, {bool parentUsesSize = false}) {
+    super.layout(constraints, parentUsesSize: parentUsesSize);
+    if (_registry == null) {
+      return;
+    }
+    if (_isRegistered) {
+      _registry!.unregisterTapRegion(this);
+    }
+    final bool shouldBeRegistered = _enabled && _registry != null;
+    if (shouldBeRegistered) {
+      _registry!.registerTapRegion(this);
+    }
+    _isRegistered = shouldBeRegistered;
+  }
+
+  @override
+  void dispose() {
+    if (_isRegistered) {
+      _registry!.unregisterTapRegion(this);
+    }
+    super.dispose();
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<Object?>('debugLabel', debugLabel, defaultValue: null));
+    properties.add(DiagnosticsProperty<Object?>('groupId', groupId, defaultValue: null));
+    properties.add(FlagProperty('enabled', value: enabled, ifFalse: 'DISABLED', defaultValue: true));
+  }
+}
+
+/// A [TapRegion] that adds its children to the tap region group for widgets
+/// based on the [EditableText] text editing widget, such as [TextField] and
+/// [CupertinoTextField].
+///
+/// Widgets that are wrapped with a `TextFieldTapRegion` are considered to be
+/// part of a text field for purposes of unfocus behavior. So, when the user
+/// taps on them, the currently focused text field won't be unfocused by
+/// default. This allows controls like spinners, copy buttons, and formatting
+/// buttons to be associated with a text field without causing the text field to
+/// lose focus when they are interacted with.
+///
+/// {@tool dartpad}
+/// This example shows how to use a `TextFieldTapRegion` to wrap a set of
+/// "spinner" buttons that increment and decrement a value in the text field
+/// without causing the text field to lose keyboard focus.
+///
+/// This example includes a generic `SpinnerField<T>` class that you can copy/paste
+/// into your own project and customize.
+///
+/// ** See code in examples/api/lib/widgets/tap_region/text_field_tap_region.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [TapRegion], the widget that this widget uses to add widgets to the group
+///    of text fields.
+class TextFieldTapRegion extends TapRegion {
```

Comment:
+1 This seems like a clean way to do this while allowing the user to put their own stuff in the same group too.